pkg/applyconfiguration/applyconfiguration_integration_test.go

```diff
@@ -0,0 +1,217 @@
+			"applyconfiguration",
+			"paths=./api/v1",
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		rt.OutputRules = genall.OutputRules{Default: output}
+
+		originalFS := os.DirFS(filepath.Join(originalCWD, cronjobDir))
+		tmpFS := os.DirFS(".")
+
+		By("Running the generator")
+		hadErrs := rt.Run()
+
+		By("Checking for generation errors")
+		Expect(hadErrs).To(BeFalse(), "Generator should run without errors")
+
+		filesInOriginal := make(map[string][]byte)
+		originalFileNames := sets.New[string]()
+		Expect(fs.WalkDir(originalFS, filepath.Join("api/v1", applyConfigurationDir), func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if d.IsDir() {
+				return nil
+			}
+
+			data, err := os.ReadFile(filepath.Join(originalCWD, cronjobDir, path))
+			if err != nil {
+				return fmt.Errorf("error reading file %s: %w", path, err)
+			}
+
+			// Record the path without the path prefix for comparison later.
+			path = strings.TrimPrefix(path, filepath.Join("api/v1", applyConfigurationDir)+"/")
+			originalFileNames.Insert(path)
+			filesInOriginal[path] = data
+			return nil
+		})).To(Succeed())
+
+		filesInOutput := make(map[string][]byte)
+		outputFileNames := sets.New[string]()
+		Expect(fs.WalkDir(tmpFS, filepath.Join("api/v1", outputPackage), func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if d.IsDir() {
+				return nil
+			}
+
+			data, err := os.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("error reading file %s: %w", path, err)
+			}
+
+			// Record the path without the path prefix for comparison later.
+			path = strings.TrimPrefix(path, filepath.Join("api/v1", outputPackage)+"/")
+			outputFileNames.Insert(path)
+			filesInOutput[path] = data
+			return nil
+		})).To(Succeed())
+
+		// // Every file should be in both sets, check for files not in both sets.
+		Expect(outputFileNames.UnsortedList()).To(ConsistOf(originalFileNames.UnsortedList()), "Generated files should match the checked in files")
+
+		for name, content := range filesInOriginal {
+			// If the output package uses a relative path we need to remove the "../" from the package name.
+			outputPackageName := strings.ReplaceAll(outputPackage, "../", "")
+
+			// Make sure the package string is correct for the newly generated content.
+			content = []byte(strings.Replace(string(content), "package applyconfiguration", fmt.Sprintf("package %s", outputPackageName), 1))
+
+			// Make sure the import paths are correct for the newly generated content.
+			content = []byte(strings.ReplaceAll(string(content), "testdata/cronjob/api/v1/applyconfiguration", filepath.Join("testdata/cronjob/api/v1", outputPackage)))
+
+			Expect(string(filesInOutput[name])).To(Equal(string(content)), "Generated files should match the checked in files, diff found in %s", name)
```

Comment:
```suggestion
			Expect(string(filesInOutput[name])).To(BeComparableTo(string(content)), "Generated files should match the checked in files, diff found in %s", name)
```
Up to you, usually this diff is better